pkg/monitor/monitorapi/types.go

```diff
@@ -266,6 +266,78 @@ func Not(filter EventIntervalMatchesFunc) EventIntervalMatchesFunc {
 	}
 }
 
+func StartedBefore(limit time.Time) EventIntervalMatchesFunc {
+	return func(eventInterval EventInterval) bool {
+		return eventInterval.From.Before(limit)
+	}
+}
+
+func EndedAfter(limit time.Time) EventIntervalMatchesFunc {
+	return func(eventInterval EventInterval) bool {
+		return eventInterval.To.After(limit)
+	}
+}
+
+func NodeUpdate(eventInterval EventInterval) bool {
+	reason := ReasonFrom(eventInterval.Message)
+	return "NodeUpdate" == reason
+}
+
+func AlertFiringInNamespace(alertName, namespace string) EventIntervalMatchesFunc {
+	return func(eventInterval EventInterval) bool {
+		return And(
+			func(eventInterval EventInterval) bool {
+				locatorParts := LocatorParts(eventInterval.Locator)
+				eventAlertName := AlertFrom(locatorParts)
+				if eventAlertName != alertName {
+					return false
+				}
+				if strings.Contains(eventInterval.Message, `alertstate="pending"`) {
```

Comment:
are pending & firing switched? or is that intentional and I'm missing it?

Reply:
nope